### DISCUSSION TITLE
Is there an example of what a custom hook function should look like?

### DISCUSSION THREAD

**Discussion by soypunk** [MrThearMan/django-signal-webhooks#65] [General]
When our CREATE hook is called I need to some conditional logic to have the webhook not sent in certain cases - is there an example of what the custom hook function should look like?

**Comment by MrThearMan** [comment]
Hi! Yes, you can define a `webhook_data(self)` method on the model the webhook is for, and raise a `WebhookCancelled` exception to cancel the hook. You can see this from the [default serializer](https://github.com/MrThearMan/django-signal-webhooks/blob/main/signal_webhooks%2Futils.py#L81).
